testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -597,116 +603,197 @@ def test_triclinic_box():
+
+        self.trajectory = self._reader(self.traj)
+        self.prec = 3
+        self.ts = self.trajectory.ts
+
+    def tearDown(self):
+        del self.tmpdir
+        del self.trajectory
+
+    @dec.slow
+    def test_offsets(self):
+        self.trajectory._read_offsets(store=True)
+        assert_array_almost_equal(self.trajectory._xdr.offsets,
+                                  self.ref_offsets,
+                                  err_msg="wrong frame offsets")
+
+        outfile_offsets = XDR.offsets_filename(self.traj)
+        with open(outfile_offsets) as f:
+            saved_offsets = {k: v for k, v in np.load(f).iteritems()}
+
+        assert_array_almost_equal(self.trajectory._xdr.offsets,
+                                  saved_offsets['offsets'],
+                                  err_msg="error saving frame offsets")
+        assert_array_almost_equal(self.ref_offsets, saved_offsets['offsets'],
+                                  err_msg="saved frame offsets don't match "
+                                  "the known ones")
+
+        self.trajectory._load_offsets()
+        assert_array_almost_equal(self.trajectory._xdr.offsets,
+                                  self.ref_offsets,
+                                  err_msg="error loading frame offsets")
+        assert_equal(saved_offsets['ctime'], os.path.getctime(self.traj))
+        assert_equal(saved_offsets['size'], os.path.getsize(self.traj))
+
+    def test_reload_offsets(self):
+        self._reader(self.traj, refresh_offsets=True)
+
+    # TODO: actually tests mismatchs, this only ensures the code-path is run
+    @dec.slow
+    def test_persistent_offsets_size_mismatch(self):
+        # check that stored offsets are not loaded when trajectory
+        # size differs from stored size
+        fname = XDR.offsets_filename(self.traj)
+        with open(fname) as f:
+            saved_offsets = {k: v for k, v in np.load(f).iteritems()}
+        saved_offsets['size'] += 1
+        with open(fname, 'w') as f:
+            np.savez(f, **saved_offsets)
+        self._reader(self.traj)
+
+    # TODO: actually tests mismatchs, this only ensures the code-path is run
+    @dec.slow
+    def test_persistent_offsets_ctime_mismatch(self):
+        # check that stored offsets are not loaded when trajectory
+        # ctime differs from stored ctime
+        fname = XDR.offsets_filename(self.traj)
+        with open(fname) as f:
+            saved_offsets = {k: v for k, v in np.load(f).iteritems()}
+        saved_offsets['ctime'] += 1
+        with open(fname, 'w') as f:
+            np.savez(f, **saved_offsets)
+        self._reader(self.traj)
+
+    # TODO: This doesn't test if the offsets work AT ALL. the old
+    # implementation only checked if the offsets were ok to set back to the old
+    # frame. But that doesn't check if any of the other offsets is potentially
+    # wrong. Basically the only way to check that would be to scan through the
+    # whole trajectory.
```

Comment:
To make sure others read this as well. It is very important to know that the old-code didn't actually work as expected